TestPidOf {procfs}

https://k8s-gubernator.appspot.com/build/kubernetes-jenkins/logs/kubernetes-test-go/15578/

Failed: TestPidOf {procfs}

```
panic: runtime error: invalid memory address or nil pointer dereference
/usr/local/go/src/runtime/panic.go:481 +0x3ff
/usr/local/go/src/testing/testing.go:467 +0x22a
/usr/local/go/src/runtime/panic.go:443 +0x521
/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/pkg/util/procfs/procfs.go:70 +0xb2
/usr/local/go/src/path/filepath/path.go:370 +0x4e8
/usr/local/go/src/path/filepath/path.go:396 +0x107
/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/pkg/util/procfs/procfs.go:104 +0x17b
/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/pkg/util/procfs/procfs_test.go:69 +0xb0
/usr/local/go/src/testing/testing.go:473 +0xdd
/usr/local/go/src/testing/testing.go:582 +0xae3
```

--- a/pkg/util/procfs/procfs.go
+++ b/pkg/util/procfs/procfs.go
@@ -65,6 +65,9 @@ func (pfs *ProcFS) GetFullContainerName(pid int) (string, error) {
 func PidOf(name string) []int {
 	pids := []int{}
 	filepath.Walk("/proc", func(path string, info os.FileInfo, err error) error {
+		if err != nil {
+			return err
+		}
 		base := filepath.Base(path)
 		// Traverse only the directories we are interested in
 		if info.IsDir() && path != "/proc" {

